setup.py

```diff
@@ -45,14 +45,32 @@
     include_dirs=include_dirs,
     libraries=['nvptxcompiler_static'],
     library_dirs=library_dirs,
-    extra_compile_args=['-Wall', '-Werror'],
+    extra_compile_args=['-Wall', '-Werror', '-std=c++11'],
 )
 
+if "RAPIDS_PY_WHEEL_VERSIONEER_OVERRIDE" in os.environ:
+    orig_get_versions = versioneer.get_versions
+
+    version_override = os.environ["RAPIDS_PY_WHEEL_VERSIONEER_OVERRIDE"]
+    if version_override == "":
+        raise RuntimeError(
+            "RAPIDS_PY_WHEEL_VERSIONEER_OVERRIDE in environment but empty"
+        )
+
+    def get_versions():
+        data = orig_get_versions()
+        data["version"] = version_override
+        return data
+
+    versioneer.get_versions = get_versions
```

Comment:
Does the shared actions workflow still set up these env vars? I know that many other packages have been migrated off versioneer.

Reply:
Good question. The short answer is no, those env vars aren't set up, but it doesn't matter here.

The longer answer: the other packages building wheels have been migrated away from versioneer and to a large extent away from setup.py as well. Almost all of their packaging metadata is in pyproject.toml now. The workflows modify that file directly since it's a well-defined config file instead of a code file. If we wanted ptxcompiler to go the same route, then we would need to update this package in that way as well. However, we don't use the shared workflows to release ptxcompiler. Since the release cycle is slower and more irregular than the rest of RAPIDS, and since the build is much simpler, we create wheels on-demand using some scripts that can be run locally. I've ensure that those scripts will still work with this versioning approach, so that should be sufficient for everything to "just work" for us.

Reply:
Thanks!